app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -123,6 +125,7 @@ def index_relation
       query = index_relation_content_unit(query)
       query = index_relation_content_view(query)
       query = index_relation_environment(query)
+      query = query.joins(:root).merge(RootRepository.deb_type.where.not(deb_errata_url: nil).or(RootRepository.yum_type)) if params[:has_errata]
```

Comment:
This is a nice idea and makes sense in the way it's used.

We have something in Katello called the RepositoryTypeManager which is aware of our various repo types and enables us to interact with repositories in a generic way based on their type definitions, take deb repos for example: https://github.com/Katello/katello/blob/master/lib/katello/repository_types/deb.rb

I'd like to see us use this manager here by defining some kind of metadata ex: `has_errata true` on the yum and deb types. that way we can avoid having to update this controller for each type of repo that will eventually support errata.
The outcome here might look like this:
```
query = index_relation_has_errata(query)

def index_relation_has_errata(query)
  query.with_type(RepositoryTypeManager.types_with_errata) if params[:has_errata]
end
```

Reply:
Would also like to suggest renaming the parameter. `has_errata` indicates to me that the repo *contains* errata, but the intention of the parameter is to list repos that *support* errata. so maybe something like 'errata_supported' or similar would be more clear

Reply:
I agree that retrieving the information about errata-capability should be provided from a central entity.
I am not sure, if the RepositoryTypeManager is a good choice for deb-repositories, because only a small subset of deb-repositories are capable to provide errata (the ones where `deb_errata_url` has been defined).
So I'd rather vote for a `errata_supported?` method in `RootRepository`, which can decide for its instance.
At least for deb-repositories. For all other RepositoryTypes it may refer to the RepositoryTypeManager after all.

Reply:
I think that'll work just as well